### PR TITLE
Update system to handle recent uW server movements; fix lost text after exclamation marks

### PR DESCRIPTION
**Pull request by RobH123**
Also handles recent uW branding changes.

Does already contain a little bit of extra unused code that will soon be used to upload the PDF files to AWS S3.

### CODE REVIEW

**Comment by RobH123** [comment]
Oh, it's automatically appended my latest commits. (I only intended the PR to cover the first 13 commits -- which get the system working as before = returning the created PDF. The later commits upload the created PDF to S3 instead.) I guess I need to learn a bit more about Github.

File: AWS-README.md [hunk 1]
```diff
@@ -11,5 +11,5 @@ exit
 
 ```bash
 docker info
-docker run --name obs-pdf --rm -p 8080:80 -i -t phopper/obs-pdf:latest bash
+docker run --name obs-pdf --rm -p 8080:80 -i -t unfoldingWord/obs-pdf:latest bash
```
Comment (jag3773) [inline AWS-README.md:14]: I see the base image on Docker hub but I don't see https://hub.docker.com/r/unfoldingword/obs-pdf there...

File: public/lib/pdf_from_dcs.py [hunk 2]
```diff
@@ -177,78 +257,131 @@ def create_pdf(self, obs_obj: OBS) -> str:
...
+            # shutil.copyfile(pdf_current_filepath, pdf_destination_filepath)
+
+            # Upload the PDF to our AWS S3 bucket
+            self.output_msg += f"{datetime.datetime.now()} => Uploading the '{obs_lang_code}' PDF file to S3 bucket…\n"
+            write_file(self.output_msg_filepath, self.output_msg)
+            cdn_s3_handler = S3Handler(bucket_name=self.cdn_bucket_name,
+                                       aws_access_key_id=self.aws_access_key_id,
+                                       aws_secret_access_key=self.aws_secret_access_key,
+                                       aws_region_name=self.aws_region_name)
+            s3_commit_key = f'obs/auto_PDFs/{pdf_desired_name}'
```
Comment (jag3773) [inline public/lib/pdf_from_dcs.py]: Per slack thread, don't put them here.  Use Option 2 listed in https://github.com/unfoldingWord-dev/door43.org/issues/389

File: resources/docker-app/README.md [hunk 3]
```diff
@@ -34,9 +38,9 @@ exit
 
 ### Run the Docker container in background and execute commands
 ```bash
-docker pull phopper/obs-pdf:latest
-docker run --name obs-pdf --rm -p 8080:80 -dit --cpus=0.5 phopper/obs-pdf:latest
-docker run --name obs-pdf -p 8080:80 -dit --cpus=0.5 --restart unless-stopped phopper/obs-pdf:latest
+docker pull unfoldingWord/obs-pdf:latest
+docker run --name obs-pdf --rm -p 8080:80 -dit --cpus=0.5 unfoldingWord/obs-pdf:latest
+docker run --name obs-pdf -p 8080:80 -dit --cpus=0.5 --restart unless-stopped unfoldingWord/obs-pdf:latest
```
Comment (jag3773) [inline resources/docker-app/README.md:43]: We don't want the `--cpus` flag on these.  If you want to reduce the CPU share that it has then I'd suggest using `--cpu-shares` and set it to a number lower than 1024, maybe 768 or 512.

File: public/lib/aws_tools/s3_handler.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+            content_type = get_mime_type(path)
+        self.bucket.put_object(
+            Key=key,
+            Body=binary,
+            ContentType=content_type,
+            CacheControl='max-age={0}'.format(cache_time)
+        )
+
+
+    # def get_object(self, key):
```
Comment (phillip-hopper) [inline public/lib/aws_tools/s3_handler.py]: I'm guessing this was copied from another project. I would just remove the methods that are commented out, that way the next person isn't trying to figure out if they are part of a feature that hasn't been implemented yet.

File: public/lib/obs/obs_classes.py [hunk 5]
```diff
@@ -222,18 +232,19 @@ def verify_all(self):
...
+        return True
+
 
 
 class OBSEncoder(JSONEncoder):
     def default(self, o):
         return o.__dict__
 
 
+
```
Comment (phillip-hopper) [inline public/lib/obs/obs_classes.py:253]: PyCharm/IntelliJ complains loudly about the extra white space.
image

File: public/lib/obs/obs_tex_export.py [hunk 6]
```diff
@@ -204,11 +211,13 @@ def get_title(self, text):
             text_dir = 'TLT'
         return '    \\startmakeup\\textdir {0}\\section{{{1}}}\\stopmakeup'.format(text_dir, text)
 
+
     @staticmethod
     def get_image(xtr, fid, res):
-        img_link = join_url_parts(OBSTexExport.api_url_jpg, res, 'obs-{0}.jpg'.format(fid))
+        img_link = join_url_parts(OBSTexExport.api_url_jpg, res, f'obs-{fid}.jpg')
```
Comment (phillip-hopper) [inline public/lib/obs/obs_tex_export.py:234]: It looks like this change does nothing.

**Review by phillip-hopper — CHANGES_REQUESTED** [review]

**Pull request closed without merging**